conans/model/profile.py

```diff
@@ -84,7 +82,7 @@ def dumps(self):
             result.append("%s: %s" % (pattern, ", ".join(str(r) for r in req_list)))
 
         result.append("[env]")
```

Comment:
Why are you leaving the empty section?

Reply:
it required to change even more tests, I preferred to do empty output [env], this can be removed in a later PR. Same as the ``no_op()`` everywhere, to reduce the diff.